### PR TITLE
Annotate and document methods of hudson.Launcher

### PR DESCRIPTION
**Pull request by oleg-nenashev**
I was investigating code around [JENKINS-38527](https://issues.jenkins-ci.org/browse/JENKINS-38527) and annotated some bits while I was around. Will push the fix in a separate PR next week.

### Changelog entries

Proposed changelog entries: Not required

### Submitter checklist

Not a functional change

### Desired reviewers

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/FilePath.java [hunk 1]
```diff
@@ -2307,13 +2307,15 @@ private void readFromTar(String name, File baseDir, InputStream in) throws IOExc
      * @since 1.89
      */
     public Launcher createLauncher(TaskListener listener) throws IOException, InterruptedException {
-        if(channel==null)
+        if(channel==null) {
             return new LocalLauncher(listener);
-        else
+        } else {
             return new RemoteLauncher(listener,channel,channel.call(new IsUnix()));
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/FilePath.java]: It will complicate backporting of the JENKINS-38527 fix, should probably revert

**Comment by oleg-nenashev** [comment]
Now FindBugs also catches something:

```
[INFO] --- findbugs-maven-plugin:3.0.4:check (findbugs) @ jenkins-core ---
[INFO] BugInstance size is 2
[INFO] Error size is 0
[INFO] Total bugs: 2
[INFO] Null passed for non-null parameter of new Launcher$RemoteLaunchCallable(List, boolean[], String[], InputStream, boolean, OutputStream, boolean, OutputStream, boolean, boolean, String, TaskListener) in hudson.Launcher$RemoteLauncher.launch(Launcher$ProcStarter) [hudson.Launcher$RemoteLauncher, hudson.Launcher$RemoteLauncher] Method invoked at Launcher.java:[line 1051]Known null at Launcher.java:[line 1051] NP_NULL_PARAM_DEREF
[INFO] new IllegalStateException(String) not thrown in hudson.Launcher$RemoteLauncher.getChannel() [hudson.Launcher$RemoteLauncher] At Launcher.java:[line 1039] RV_EXCEPTION_NOT_THROWN
[INFO] 
```

One is my mistake

**Review by stephenc — APPROVED** [review]
Looks ok, nothing obviously wrong

**Comment by oleg-nenashev** [comment]
@reviewbybees done. Should be merged with squash to simplify backporting of JENKINS-38527

**Pull request merged**